crates/goose/src/providers/utils.rs

```diff
@@ -243,10 +253,10 @@ pub async fn handle_response_google_compat(response: Response) -> Result<Value,
             );
             Err(ProviderError::RequestFailed(format!("Request failed with status: {}. Message: {}", final_status, error_msg)))
         }
-        _ if final_status.is_server_error() => {
-            Err(ProviderError::ServerError(format!("{:?}", payload)))
+        StatusCode::TOO_MANY_REQUESTS => {
+            Err(ProviderError::RateLimitExceeded(format!("{:?}", payload)))
         }
-        StatusCode::INTERNAL_SERVER_ERROR | StatusCode::SERVICE_UNAVAILABLE => {
+        _ if final_status.is_server_error() => {
```

Comment:
ok so this will be a catch all - and even if TOO_MANY_REQUESTS triggers above it will retry?

Reply:
I just reverted thew whole thing. sorry for stealing your attention to an unrelated area of code.